File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int buyAt = 0;
        for (int i = 0; i < prices.length; i++){
            int p = prices[i];
            if (prices[buyAt] > p){
                buyAt = i;
            }
            int current_profit = p - prices[buyAt];
            if (current_profit > profit){
                profit = current_profit;
            }
        }

        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int res = 0;
        int cur = int.MaxValue;

        foreach (int p in prices){
            cur = (p < cur) ? p:cur;
            res = (p-cur > res) ? p-cur : res;
        }
        return res;
    }
}